ui/src/app/icons/edit/component.tsx

```diff
@@ -0,0 +1,73 @@
+import {Component, OnInit} from '@angular/core';
+
+import * as React from 'react';
+import * as ReactDOM from 'react-dom';
+
+interface EditComponentState { style: object; }
+
+class EditComponentReact extends React.Component<{}, EditComponentState> {
```

Comment:
opt: I suspect this may be simple enough to use the functional form of a react component to minimize boilerplate: https://reactjs.org/docs/components-and-props.html

Reply:
I had considered that, and can easily be persuaded to do that. I was trying to be consistent with our typing of props and state in other conversions - will give it some more thought.

Reply:
I ended up not doing this - while it compiled and ran fine, the IDE complains about typing when trying to access state variables.